Login

lorenzoz23 opened the pull request:
yeah i had some fun...

also left a comment to change the background so you can see how it actually looks like when I'm not trying to meme

but real talk it actually kinda looks cool

lorenzoz23 committed:
trying out colors idk

lorenzoz23 committed:
ur gonna hate me honestly

lorenzoz23 committed:
added an out

katebrune commented:
about gist section
- we will have to work on this over time to figure out what we want to say so i think its fine for now, but also the clock is giving me anxiety, but the component is cool

login section
- the galaxy ... has been discussed, i think we can figure out a gradient, but also if it was smaller, like closer around the section 
- when on the sign in, the 'sign up' is highlighted green and vice versa, i think it would look nicer if it was green when it was selected and the other one was the darker one
- on 'sign in' the remember me isn't switchable yet
- on 'sign up' center the create button
- i think it would look nice if we had 'gist' up on the left or something and 'login' or something where gist is on the login section
- we could also do some sort of gradient or pattern in the back and have some opacity over the login section, just spit ballin',

overall though, good start, hell ya

katebrune commented on frontend/gist-frontend/src/gist-login/LoginPageComponent.tsx:
```diff
@@ -8,13 +8,21 @@ import { Box, Grommet, Tab, Tabs, Heading } from "grommet";
 const theme = {
```
eventually move theme into it's own file

lorenzoz23 replied:
done and done

katebrune commented on frontend/gist-frontend/src/gist-login/UserLoginComponent.tsx:
```diff
@@ -1,5 +1,22 @@
-import { Form, FormField, Button, CheckBox } from "grommet";
+import { Form, FormField, Button, CheckBox, Box, Grommet } from "grommet";
+
+const theme = {
```
move theme to separate file eventually

lorenzoz23 replied:
smashed it

katebrune reviewed (changes requested):
maybe think about making a component just to hold login stuff (userlogincomponent, createusercomponent) and then have a separate component that is the whole page with just the aboutgistcomponent and logincomponent

lorenzoz23 committed:
big changes comin ur way

lorenzoz23 committed:
updated to include most, if not all, suggestions from pull request

lorenzoz23 committed:
messed up with a button, but it's all good now

katebrune commented on frontend/gist-frontend/src/gist-login/LoginOptionsComponent.tsx:
```diff
@@ -0,0 +1,69 @@
+  };
+
+  render() {
+    let loginTitle: String = this.getLoginTitle();
```
could do : 

`const {isLoginTabActive} = this.state`
`const title : string = isLoginTabActive? 'login' : 'create' `

to get rid of getLoginTitle() function

lorenzoz23 replied:
did, done, smash

katebrune commented on frontend/gist-frontend/src/gist-login/UserLoginComponent.tsx:
```diff
@@ -1,29 +1,118 @@
+import { Form, FormField, Button, CheckBox, Box } from "grommet";
+
+interface IUserLoginComponentState {
+  checked: boolean;
```
what is checked? make more specific

lorenzoz23 replied:
yep

katebrune reviewed (changes requested):
style comments
 - the bar highlighting for 'sign up' and 'sign in' is still backwards
- i like the login button being locked until you enter stuff, nice touch there
- mess around with some different styles for profile picture upload, also need the option to not upload one and have a default

project structure
- instead of gist-login as the folder name, do components, its just standard for react
- throw theme.tsx in a helpers folder, we will probably end up with a lot of those

forms
- as much as formik sucks, we might want to look into because the create user is gonna have a lot of fields to check

lorenzoz23 committed:
implemented pr changes and added image modal pop up

lorenzoz23 committed:
added functionality for updating default image choice

lorenzoz23 committed:
finally fixed that tab coloring issue

lorenzoz23 commented:
So, yeah... I need to start keeping better track of what I do before every major push to my branch. Also, better commit messages.

Anywho...

I am pretty sure I finally implemented all your suggestions from the past two reviews. I'm looking into formik and yup validation right now to see how that will all work out. Admittedly, it's been a really long time since I've done that so it might take some time on my end.

As far as stuff that I added for funsies goes, I made it so that the remember me checkbox is also disabled until both login and password has input and I added a modal pop-up for selecting a profile picture. The modal has three options of image importing via filesystem, url, or selecting one of the three default svgs  that I chose. The latter is implemented with a drop-down select component while the filesystem button will open up the filesystem and the url button will reveal a hidden text field to input in the url. When the user chooses one of the three user icons, I have it so that it updates the one shown in the form. 

Yeah, I think that's it, boss. It's 5:30 and I've got children to go home to. Please, let me live.

bye

katebrune commented on frontend/gist-frontend/src/Components/AboutGistComponent.tsx:
```diff
@@ -0,0 +1,28 @@
+export default class AboutGistComponent extends React.Component {
+  render() {
+    return (
+      <div>
```
instead of `<div>...component stuff...</div>`, you can use `<>...component stuff... </>` for the outside parts of react components

lorenzoz23 replied:
Oh you...

katebrune commented on frontend/gist-frontend/src/Components/CreateUserComponent.tsx:
```diff
@@ -0,0 +1,142 @@
+    showPictureModal: false,
+    image: "User"
+  };
+  onImageButtonClick = () => {
```
we should rename this to something more like select image, i kept looking around for an 'Image Button' like a button with an image and was confused

lorenzoz23 replied:
Wasn't crazy about that name anyway; some weird names float around my head that early in the morning.

katebrune commented on frontend/gist-frontend/src/Components/CreateUserComponent.tsx:
```diff
@@ -0,0 +1,142 @@
+
+interface ICreateUserComponentState {
+  showPictureModal: boolean;
+  image: string;
```
instead of `state{image:string}` could we do `state{image : JSX.Element}` and just store the element here? i think it would get rid of a lot of the `if(image === "UserFemale")` etc. We don't have images set up yet on the backend, so we don't have to worry about storing an image string yet. We could just have the icons as the only choices for now.

lorenzoz23 replied:
Yeah, this makes a lot of sense. I previously was doing that, but for some reason reverted to strings and have no idea why.

katebrune reviewed (commented):
style comments
- i like the popup for profile picture, only thing i have to say is i feel like it should be rounded to fit the theme of the main login page
- nit picky as hell but i want some space between the profile picture guy and the select image button
- we should look into some gradient stuff, i'm not crazy about the white background for the login area but it works for now

overall, the code is looking pretty clean and i feel like we are almost done with this, hell ya, good job

lorenzoz23 committed:
updated pr to include fixes from comments

lorenzoz23 commented:
I tried to resolve all issues you had with the previous pull request, but alas, couldn't catch em all. 

- Added a bit of space between the user icon and the select image button

- Added a gradient (still not sold on chosen colors, but I tried to make it as light as I could to better show the text) but also need to update the color of the text in all the form fields and buttons to a darker color so we can see it. 

- Changed the div tags to <> in all component classes

- Renamed the image button to onSelectImageClicked(), or something like that, I forget...

- I remembered why I couldn't just store the user icon as a jsx element and it was because of the fact that I need the options of the select dropdown to be strings and not icons. Inserting icons to the select component makes everything look weird but strings render fine, so I kept the string variable and just added a helper method that takes in a color and size variable to serve both types of icons we need: singular icon above the button and icons for each item in the select. This all resulted in less code even though it wasn't exactly what you hoped for, but it's the best I can do for now with how thi framework works.

One last thing...

- I could not get the modal pop up to be rounded because the round attribute that grommet provides does not actually round the edges of the box itself, but of the border within it. So, when I did have the round attribute inside the box tag, the border inside the box rendered with rounded edges revealing the white background color of the box between the edge of the box and edge of the border. The only reason this worked for the create/login box to the right of the screen is because I set the background color to be the same green that the rest of the screen has as a background, so everything just blended together nicely; I can't do that with this obviously because I have the box centered with one color on the right and a different one to the left. I could have the pop up render to the left perhaps and set the background to that green color but idk how well that will look having a pop up render to the left and not center. Personally, I think it looks fine as is currently, even though I do agree with you when you say it should be rounded. We can probably just leave it for now.

lorenzoz23 commented:
So yeah, still have to update the font above the gradient and probably change the gradient all together because I still don't know if I'm happy with it. Hopefully, this gives you an idea of what a gradient will look like within the app though.

katebrune commented on frontend/gist-frontend/src/Components/LoginOptionsComponent.tsx:
```diff
@@ -0,0 +1,62 @@
+    const { isLoginTabActive } = this.state;
+    const loginTitle: String = isLoginTabActive ? "login" : "create";
+    return (
+      <Box
```
played around with a style if you wanna check it out

`<Box
        border={{
          color: "#1BD7BB",
          size: "large",
          style: "ridge",
          side: "all"
        }}
        round={true}
        width="large"
        height="large"
        background={{
          color: "#FFFFFF",
          opacity: "medium"
        }}
        margin={{
          right: "large",
          top: "large"
        }}
        basis="auto"
        elevation="xlarge"
        align="center"
        justify="center"
        overflow={{ vertical: "auto" }}
      >`

katebrune commented on frontend/gist-frontend/src/Components/LoginPageComponent.tsx:
```diff
@@ -0,0 +1,33 @@
+  render() {
+    return (
+      <Grommet theme={this.props.theme} full>
+        <Box fill background="brand">
```
played around with a theme if you wanna check it out

`<Box
          fill
          background="linear-gradient(160deg,#a8f4d3,#7de3e1,#6cceea,#7cb5e6,#94a7e1,#aba5e5,#c5a1e4,#de9ddd)"
        >`

katebrune reviewed (commented):
added a few comments of where i put some stuff when i was playing around with the style with gradients if ya wanna check it out
